service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -107,30 +107,79 @@ sudo cp terra /usr/bin/terra
 sudo -u "${JUPYTER_USER}" sh -c "terra config set browser MANUAL"
 # Set the CLI terra server based on the terra server that created the GCP notebook retrieved from
 # the VM metadata, if set.
-readonly TERRA_SERVER=$(get_metadata_value "instance/attributes/terra-cli-server")
+readonly TERRA_SERVER="$(get_metadata_value "instance/attributes/terra-cli-server")"
 if [[ -n "${TERRA_SERVER}" ]]; then
   sudo -u "${JUPYTER_USER}" sh -c "terra server set --name=${TERRA_SERVER}"
 fi
 
 # Log in with app-default-credentials
 sudo -u "${JUPYTER_USER}" sh -c "terra auth login --mode=APP_DEFAULT_CREDENTIALS"
 
+####################################
+# Shell and notebook environment
+####################################
+
 # Set the CLI terra workspace id using the VM metadata, if set.
-readonly TERRA_WORKSPACE=$(get_metadata_value "instance/attributes/terra-workspace-id")
+readonly TERRA_WORKSPACE="$(get_metadata_value "instance/attributes/terra-workspace-id")"
 if [[ -n "${TERRA_WORKSPACE}" ]]; then
   sudo -u "${JUPYTER_USER}" sh -c "terra workspace set --id=${TERRA_WORKSPACE}"
 fi
 
+# Set variables into the .bash_profile such that they are available
```

Comment:
Could it be organized like this

```
# Set variables into the .bash_profile such that they are available
# to terminals, notebooks, and other tools
# Leonardo environment variables are at # https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/resources/init-resources/startup.sh

readonly OWNER_EMAIL="$(
  sudo -u "${JUPYTER_USER}" sh -c "terra workspace describe --format=json" | \
  jq --raw-output ".userEmail")"
echo "export OWNER_EMAIL='${OWNER_EMAIL}'" >> "/home/${JUPYTER_USER}/.bash_profile"

# Leonardo sets GOOGLE_PROJECT.
readonly GOOGLE_PROJECT="$(
  sudo -u "${JUPYTER_USER}" sh -c "terra workspace describe --format=json" | \
  jq --raw-output ".googleProjectId")"
echo "export GOOGLE_PROJECT='${GOOGLE_PROJECT}'" >> "/home/${JUPYTER_USER}/.bash_profile"

# Also set GOOGLE_CLOUD_PROJECT in case people want that instead of GOOGLE_PROJECT
echo "export GOOGLE_CLOUD_PROJECT='${GOOGLE_PROJECT}'" >> "/home/${JUPYTER_USER}/.bash_profile"

# Leonardo sets PET_SA_EMAIL.
readonly PET_SA_EMAIL="$(
  sudo -u "${JUPYTER_USER}" sh -c "terra auth status --format=json" | \
  jq --raw-output ".serviceAccountEmail")"
echo "export PET_SA_EMAIL='${PET_SA_EMAIL}'" >> "/home/${JUPYTER_USER}/.bash_profile"

#  Also set GOOGLE_SERVICE_ACCOUNT_EMAIL in case people want that instead of PET_SA_EMAIL.
echo "export GOOGLE_SERVICE_ACCOUNT_EMAIL='${PET_SA_EMAIL}'" >> "/home/${JUPYTER_USER}/.bash_profile"

```

Reply:
I personally like the separation of the variables into the two separate categories (those set for consistency with Leonardo and those set for consistency with the Terra CLI).

Reply:
Now that I understand things better, SG.

Just checking -- if you run `terra` cli in notebook, it's ok that Leo variables aren't set, right?

Can you just clearly mark which 3 variables should be kept in sync with CLI, and which 3 should not?

```
# We have new-style variables (eg GOOGLE_CLOUD_PROJECT) which are set here and CLI (terra app execute env).
# We also support Leonardo variables (eg GOOGLE_PROJECT). Those are only set here and NOT CLI.
# Those are only for running notebooks ported from Leonardo. We encourage users to use the
# new-style variables.
```